index.js

```diff
@@ -307,5 +307,93 @@ class Stack {
   }
 }
 
+class SinglyLinkedList {
+
+	/**
+	 * @What instantiates the list.
+	 */
+	constructor() {
+		this.size = 0;
+		this.head = null;
+    this.tail = null;
+  }
+  
+  /**
+   * This method is specific to the class.
+   * @What creates a newNode for the list. Sets the next as null.
+   * @Params {js object or a primitive datatype} data 
+   */
+  newListNode(data) {
+    let node = {
+      'data': data,
+      'next': null
+    }
+    return node;
+  }
+
+	/**
+	 * @what Gets singly LL size
+	 */
+	getSize() {
+		return this.size;
+	}
+
+	/**
+	 * @What Adds a node to the beginning of the list.
+	 * @Params {value of the node} val 
+	 */
+	addFirst(val) {
+		var node = this.newListNode(val);
+		if (this.head === null) {
```

Comment:
if(!this.head) would be equivalent to current code. Please change it to this format as it is more clean